### PR TITLE
add azure openai

### PR DESCRIPTION
**Pull request by ruanwz**
Add following env:
AZURE_OPENAI_API_VERSION
AZURE_OPENAI_API_KEY
AZURE_OPENAI_API_BASE
then use as following code:
engine = get_engine("azure-gpt-35-turbo")

### CODE REVIEW

File: textgrad/engine/openai.py [hunk 1]
```diff
@@ -96,3 +96,30 @@ def generate_with_messages(self, messages, temperature=0, max_tokens=2000, top_p
...
+        system_prompt=ChatOpenAI.DEFAULT_SYSTEM_PROMPT,
+        **kwargs):
+        """
+        :param model_string:
+        :param system_prompt:
+        """
+        root = platformdirs.user_cache_dir("textgrad")
+        cache_path = os.path.join(root, f"cache_azure_{model_string}.db")  # Changed cache path to differentiate from OpenAI cache
+
+        super().__init__(cache_path=cache_path, system_prompt=system_prompt, **kwargs)
```
Comment (kgourgou) [inline textgrad/engine/openai.py:112]: Won't this call `ChatOpenAI.__init__()` and so redefine `cache_path` as         `cache_path = os.path.join(root, f"cache_openai_{model_string}.db")`?

File: textgrad/engine/openai.py [hunk 2]
```diff
@@ -96,3 +96,30 @@ def generate_with_messages(self, messages, temperature=0, max_tokens=2000, top_p
...
+
+        self.system_prompt = system_prompt
+        api_version = os.getenv("AZURE_OPENAI_API_VERSION", "2023-07-01-preview")
```
Comment (kgourgou) [inline textgrad/engine/openai.py:115]: Maybe you don't want to hardcode `"2023-07-01-preview"` there but define as `FALLBACK_AZURE_OPENAI_API_VERSION` at the top or something.

File: textgrad/engine/openai.py (hunk 1 above)
Comment (ruanwz) [reply textgrad/engine/openai.py:112]: thanks! may need to change ChatOpenAI too. But I don't want to introduce too many changes. Please advice how to fix it.

File: textgrad/engine/openai.py (hunk 2 above)
Comment (ruanwz) [reply textgrad/engine/openai.py:115]: I have added 3 env, maybe 4 is too many.

File: textgrad/engine/openai.py (hunk 1 above)
Comment (kgourgou) [reply textgrad/engine/openai.py:112]: Yep; needs to be fixed separately. 

https://github.com/zou-group/textgrad/pull/20#issuecomment-2181754251

Comment (beansrowning) [reply textgrad/engine/openai.py:112]: FWIW I use `super(ChatOpenAI, self).__init__(cache_path=cache_path)` locally.

**Comment by vinid** [comment]
this is awesome! thanks so much for this! 

can you resolve conflicts and write a docstring for the new class? you can add an explanation about the env variables!

Ty!

**Review by vinid — APPROVED** [review]

**Pull request merged**